src/Controllers/group.controllers.ts

```diff
@@ -77,61 +77,51 @@ interface GetGroupQueryParams {
 export const getGroup = async (req: Request<GetGroupQueryParams>, res: Response) => {
   const { pid } = req.params;
 
-  try {
-    const group: {
-      pid: string;
-      name: string;
-      organisation: { pid: string; name: string };
-      admins?: { pid: string; name: string }[];
-      participants?: { pid: string }[];
-    } | null = await prisma.group.findUnique({
-      where: { pid },
-      select: req.auth?.isAuthenticated
-        ? {
-            pid: true,
-            name: true,
-            organisation: { select: { pid: true, name: true } },
-            admins: { select: { pid: true, name: true } },
-            participants: { select: { pid: true } },
-          }
-        : basicGroup,
-    });
-
-    if (!group) {
-      return res.status(404).json(generateError(`The group with ID '${pid}' could not be found`));
-    }
-
-    return res.status(200).json({
-      type: "success",
-      payload: {
-        group: {
-          ...group,
-          organisation: {
-            ...group.organisation,
-            _links: [{ rel: "self", type: "GET", href: `/api/organisation/${group.organisation.pid}` }],
-          },
-          ...(req.auth?.isAuthenticated
-            ? {
-                admins: group.admins?.map((admin) => ({
-                  ...admin,
-                  _links: [{ rel: "self", type: "GET", href: `/api/admins/${admin.pid}` }],
-                })),
-                participants: group.participants?.map((participant) => ({
-                  ...participant,
-                  _links: [{ rel: "self", type: "GET", href: `/api/participant/${participant.pid}` }],
-                })),
-              }
-            : {}),
+  const group: {
+    pid: string;
+    name: string;
+    organisation: { pid: string; name: string };
+    admins?: { pid: string; name: string }[];
+    participants?: { pid: string }[];
+  } | null = await prisma.group.findUnique({
+    where: { pid },
+    select: req.auth?.isAuthenticated
+      ? {
+          pid: true,
+          name: true,
+          organisation: { select: { pid: true, name: true } },
+          admins: { select: { pid: true, name: true } },
+          participants: { select: { pid: true } },
+        }
+      : basicGroup,
+  });
+  if (!group) {
+    throw new NotFoundError("group", pid);
+  }
+  return res.status(200).json({
+    type: "success",
+    payload: {
+      group: {
+        ...group,
+        organisation: {
+          ...group.organisation,
+          _links: [{ rel: "self", type: "GET", href: `/api/organisation/${group.organisation.pid}` }],
         },
+        ...(req.auth?.isAuthenticated
+          ? {
+              admins: group.admins?.map((admin) => ({
+                ...admin,
+                _links: [{ rel: "self", type: "GET", href: `/api/admins/${admin.pid}` }],
```

Comment:
Is this the best way of adding restful links to the response, or is there something that can make that easier so its consistant

Reply:
Good that you mentioned it. Actually I'm a bit overwhelmed what it means and why it is. Probably @stephan418 can say more about it. 🦐

Reply:
We could of course add a function that generates those links (simillar to the createMediaLinks function in media.controller.ts). We stopped adding those links for the most part tho, so I simply didn't care at the time.

We could discuss if we start adding links again (and therefore implement HATEOAS) or if we just ignore it and possibly remove old ones)